packetbeat/docs/packetbeat-geoip.asciidoc

```diff
@@ -0,0 +1,165 @@
+[[packetbeat-geoip]]
+== Exporting GeoIP Information 
+
+You can use Packetbeat along with the
+{plugindoc}/ingest-geoip.html[ingest geoIP processor plugin] in Elasticsearch 
+to export geographic location information about source IPs for incoming HTTP
+requests. Then you can use this info to visualize the location of your
+clients on a map in Kibana.
+
+Prior to version 5.0, Packetbeat provided a `geoip` configuration option for
+exporting geoIP information about the source IPs.  Starting with 5.0, the
+`geoip` configuration option in Beats is deprecated in favor of using the
+ingest geoIP processor plugin. This plugin adds information about the
+geographical location of IP addresses, based on data from the Maxmind GeoLite2
+City Database. Using this plugin provides you with greater flexibility and
+control over the information that's returned by the geoIP lookup. 
+
+//REVIEWERS: trying to put some words around why the plugin is better, but I'm not sure if this is completely true. ^^
+
+NOTE: If your use case involves using Logstash, you can use the
+{logstashdoc}/plugins-filters-geoip.html[GeoIP filter] available in Logstash
+instead of using the ingest plugin. However, using the ingest plugin is the
+simplest approach when you don't require the additional processing power of
+Logstash.
+
+[float]
+[[packetbeat-configuring-geoip]]
+=== Configuring the Ingest GeoIP Processor Plugin
+
+To configure Packetbeat and the ingest geoIP processor plugin:
+
+1. {plugindoc}/ingest-geoip.html[Install the ingest geoIP processor plugin].
+After installing the plugin, remember to restart the node.
+
+2. Define an ingest node pipeline that uses the `geoip` processor. For example,
+you could define the following pipeline in a file named `geoip-pipeline.json`:
++
+[source,json]
+-------------------------------------------------------------------------------
+{
+  "description" : "Add geoip info",
+  "processors" : [
+    {
+      "geoip" : {
+        "field" : "client_ip",
+        "ignore_failure" : true
+      }
+    }
+  ]
+}
+-------------------------------------------------------------------------------
++
+Where `client_ip` is the output field in Packetbeat that contains the IP
+address of the client. You set `ignore_failure` to `true` so that the pipeline
+does not fail when it processes events that don't have a `client_ip` field. See 
+{plugindoc}/using-ingest-geoip.html[Using the Geoip Processor in a Pipeline] for
+more options. 
+
+3. Add the pipeline to Elasticsearch. For example:
++
+[source,shell]
+-------------------------------------------------------------------------------
+curl -XPUT 'http://localhost:9200/_ingest/pipeline/geoip-info' -d@geoip-pipeline.json
+-------------------------------------------------------------------------------
```

Comment:
I prefer your streamlined approach. I was trying to be consistent with how we describe using ingest node elsewhere, but I think it's OK for users to see a couple of different ways to define pipelines and to know that the syntax is a bit easier if you use the console in Kibana.